library/std/src/io/buffered/bufreader/buffer.rs

```diff
@@ -97,6 +97,25 @@ impl Buffer {
         self.pos = self.pos.saturating_sub(amt);
     }
 
+    pub fn read_more(&mut self, mut reader: impl Read) -> io::Result<()> {
+        let mut buf = BorrowedBuf::from(&mut self.buf[self.pos..]);
+        let old_init = self.initialized - self.pos;
+        unsafe {
+            buf.set_init(old_init);
+        }
+        reader.read_buf(buf.unfilled())?;
+        self.filled += buf.len();
+        self.initialized += buf.init_len() - old_init;
+        Ok(())
+    }
+
+    pub fn backshift(&mut self) {
```

Comment:
would be nice to have some docs for these fn? thanks

Reply:
done